lib/routes/utils/request_helper.js

```diff
@@ -80,8 +80,56 @@ function shouldSendVerifyAccountEmail(account, request) {
   return (!context || !!sendEmailIfUnverified) && !account.emailVerified
 }
 
+/**
+ * Returns whether or not to use signin unblock feature on a request.
+ *
+ * @param account
+ * @param config
+ * @param request
+ * @returns {boolean}
+ */
+function shouldEnableSigninUnblock(account, config, request) {
```

Comment:
I've been watching that PR, and was glad, because I felt similarly that we were growing these "is this feature enabled for this user" instances, and probably could DRY a little better.